osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyReverseArrow.cs

```diff
@@ -80,36 +80,25 @@ private void onHitObjectApplied(DrawableHitObject drawableObject)
             drawableRepeat.DrawableSlider.OverlayElementContainer.Add(proxy);
         }
 
-        private void updateStateTransforms(DrawableHitObject hitObject, ArmedState state)
+        protected override void Update()
         {
-            const double duration = 300;
-            const float rotation = 5.625f;
+            base.Update();
 
-            switch (state)
+            if (Time.Current >= drawableRepeat.HitStateUpdateTime && drawableRepeat.State.Value == ArmedState.Hit)
+            {
+                double animDuration = Math.Min(300, drawableRepeat.HitObject.SpanDuration);
+                arrow.Scale = new Vector2(Interpolation.ValueAt(Time.Current, 1, 1.4f, drawableRepeat.HitStateUpdateTime, drawableRepeat.HitStateUpdateTime + animDuration, Easing.Out));
+            }
+            else
             {
-                case ArmedState.Idle:
-                    if (shouldRotate)
-                    {
-                        InternalChild.ScaleTo(1.3f)
-                                     .RotateTo(rotation)
-                                     .Then()
-                                     .ScaleTo(1f, duration)
-                                     .RotateTo(-rotation, duration)
-                                     .Loop();
-                    }
-                    else
-                    {
-                        InternalChild.ScaleTo(1.3f).Then()
-                                     .ScaleTo(1f, duration, Easing.Out)
-                                     .Loop();
-                    }
-
-                    break;
-
-                case ArmedState.Hit:
-                    double animDuration = Math.Min(300, drawableRepeat.HitObject.SpanDuration);
-                    InternalChild.ScaleTo(1.4f, animDuration, Easing.Out);
-                    break;
+                const double duration = 300;
+                const float rotation = 5.625f;
+
+                double loopCurrentTime = (Time.Current - drawableRepeat.AnimationStartTime.Value) % duration;
+
+                if (shouldRotate)
+                    arrow.Rotation = Interpolation.ValueAt(loopCurrentTime, rotation, -rotation, 0, duration);
+                arrow.Scale = new Vector2(Interpolation.ValueAt(loopCurrentTime, 1.3f, 1, 0, duration));
```

Comment:
Easing specs disappeared here. Should be `Easing.Out` if `shouldRotate` is false.